### PR TITLE
seafile-syncing-client: Add version 7.0.9

### LINKED ISSUE #4949: [Request] Add seafile
Project Homepage: https://www.seafile.com/en/download/

Since there is a `nextcloud` client, I'd thought there might be also one for seafile

### PR DESCRIPTION
**Pull request by jfastnacht**
close #4949

### CODE REVIEW

File: bucket/seafile-syncing-client.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    "description": "Seafile desktop syncing client",
+    "version":  "7.0.9",
+    "license":  "Apache-2.0",
+    "url":  "https://download.seadrive.org/seafile-7.0.9-en.msi",
+    "hash":  "3f837823c23403ad2d5aec29f81028aaf1e86055e2fa45a8b67fcf94f59ddade",
+    "extract_dir":  "PFiles\\Seafile\\bin",
+    "post_install": "[Environment]::SetEnvironmentVariable(\"SEAFILE_DATA_DIR\",\"$DIR\\data\",\"User\")",
+    "uninstaller": {
+        "script": "[Environment]::SetEnvironmentVariable(\"SEAFILE_DATA_DIR\",$null,\"User\")"
+    },
```
Comment (Ash258) [inline bucket/seafile-syncing-client.json:12]: `env_set` https://github.com/lukesampson/scoop/wiki/App-Manifests

File: bucket/seafile-syncing-client.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "persist": "data",
```
Comment (Ash258) [inline bucket/seafile-syncing-client.json:13]: Persist after shorcuts

File: bucket/seafile-syncing-client.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "shortcuts": [
+        [
+            "seafile-applet.exe",
+            "Seafile Desktop Syncing Client"
+        ]
+    ],
+    "checkver": {
+        "url": "https://www.seafile.com/en/download/",
```
Comment (Ash258) [inline bucket/seafile-syncing-client.json:21]: Use changelog https://download.seafile.com/published/seafile-manual/changelog/client-changelog.md

**Review by Ash258 — CHANGES_REQUESTED** [review]
Reformat the manifest with checkver/formatjson utility

File: bucket/seafile-syncing-client.json (hunk 3 above)
Comment (jfastnacht) [reply bucket/seafile-syncing-client.json:21]: Is `checkver` rendering the page? If not, the changelog URL itself won't help.

Should I use https://download.seafile.com/api/v2.1/wikis/seafile-manual/content/?p=%2Fchangelog%2Fclient-changelog.md&format=json instead with `(\d+\.\d+\.\d+)` as the regex?

File: bucket/seafile-syncing-client.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+    "homepage": "https://www.seafile.com/",
+    "description": "Seafile desktop syncing client",
+    "version": "7.0.9",
+    "license": "Apache-2.0",
+    "url": "https://download.seadrive.org/seafile-7.0.9-en.msi",
+    "hash": "3f837823c23403ad2d5aec29f81028aaf1e86055e2fa45a8b67fcf94f59ddade",
+    "extract_dir": "PFiles\\Seafile\\bin",
+    "env_set": {
+        "SEAFILE_DATA_DIR": "$DIR\\data"
```
Comment (Ash258) [inline bucket/seafile-syncing-client.json:10]:
```suggestion
        "SEAFILE_DATA_DIR": "$dir\\data"
```

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/seafile-syncing-client.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    "env_set": {
+        "SEAFILE_DATA_DIR": "$dir\\data"
+    },
+    "shortcuts": [
+        [
+            "seafile-applet.exe",
+            "Seafile Desktop Syncing Client"
+        ]
+    ],
+    "persist": "data",
```
Comment (Ash258) [inline bucket/seafile-syncing-client.json:18]: What is this for?? The folder is not used

image

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/seafile-syncing-client.json (hunk 5 above)
Comment (jfastnacht) [reply bucket/seafile-syncing-client.json:18]: I'm not sure, but there's another environment variable which probably handles that directory as well. I'll implement it.

Comment (jfastnacht) [reply bucket/seafile-syncing-client.json]: Good find, haven't seen that before. It is creating a `C:\Users\<username>\ccnet` folder no matter what, but now the important `seafile.ini` is managed within the scoop directory.

Comment (Ash258) [reply bucket/seafile-syncing-client.json]: Then why do you nned the first environment variable?

Comment (jfastnacht) [reply bucket/seafile-syncing-client.json]: Apparently it's not needed if you set the `CCNET_CONF_DIR` env variable and the data directory in the `seafile.ini`. Anyways, the code states that it uses the environment variable at some point, even though it seems to ignore it here: https://github.com/haiwen/seafile-client/blob/afc17fc2a8022a82b90569f9136d21c47adb4843/src/configurator.cpp#L198-L203

I'd rather leave it be, the `SEAFILE_DATA_DIR` will not be used by any other application than Seafile though, so it won't hurt.

Comment (Ash258) [reply bucket/seafile-syncing-client.json]: Remove data dir environment variable. No need to flood users environment with junk variables.

File: bucket/seafile-syncing-client.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    "env_set": {
+        "CCNET_CONF_DIR": "$dir\\ccnetconfdir"
+    },
+    "post_install": "Set-Content \"$dir\\ccnetconfdir\\seafile.ini\" \"$dir\\data\"",
+    "shortcuts": [
+        [
+            "seafile-applet.exe",
+            "Seafile Desktop Syncing Client"
+        ]
+    ],
```
Comment (Ash258) [inline bucket/seafile-syncing-client.json:18]:
```suggestion
    "pre_install": [
        "if (!(Test-Path \"$persist_dir\\ccnetconfdir\\seafile.ini\")) {",
        "    ensure \"$dir\\ccnetconfdir\" | Out-Null",
        "    $cur = current_dir \"$dir\"",
        "    Set-Content \"$dir\\ccnetconfdir\\seafile.ini\" \"$cur\\data\" -Encoding Ascii",
        "}"
    ],
    "shortcuts": [
        [
            "seafile-applet.exe",
            "Seafile Desktop Syncing Client"
        ]
    ],
    "env_set": {
        "CCNET_CONF_DIR": "$dir\\ccnetconfdir"
    },
```

**Review by Ash258 — CHANGES_REQUESTED** [review]
ccnet folder still created in userprofile

A

User still need to choose the folder manually

B

**Review by jasongodev — CHANGES_REQUESTED** [review]
I checked Seafile and saw that there's a new version 8 that only works for Windows 10 and above. It has significant changes and improvement over Version 7.

I suggest to update this manifest to detect and autoupdate to version 8.

I also suggest to add a manifest in the Versions bucket for version 7 so that users in Windows 7 can still use Seafile version 7 using the versions bucket.

**Review by HUMORCE — APPROVED** [review]
Tested.

**Pull request merged**